oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -492,10 +503,10 @@ private static boolean isAwsCredential(Map<String, Object> credentialSource) {
   protected AccessToken exchangeExternalCredentialForAccessToken(
       StsTokenExchangeRequest stsTokenExchangeRequest) throws IOException {
     // Handle service account impersonation if necessary.
-    // Internal override takes priority.
-    if (impersonatedCredentialsOverride != null) {
-      return impersonatedCredentialsOverride.refreshAccessToken();
-    } else if (impersonatedCredentials != null) {
+    if (this.isImpersonationEnabled() && this.impersonatedCredentials == null) {
+      this.impersonatedCredentials = this.buildImpersonatedCredentials();
+    }
+    if (impersonatedCredentials != null) {
       return impersonatedCredentials.refreshAccessToken();
     }
```

Comment:
This can look easier to understand and simpler if combined. Or does it mean that impersonatedCredentials can be provided with isImpersonationEnabled == false?

Reply:
Not sure exactly what you are implying with combined? you mean adding the this.impersonatedCredentials check  into the isImpersonationEnabled method? I can do that, would just rename the method shouldBuildImpersatedCredentials I guess. The idea of the check here is to only call the build method if we haven't already instantiated the impersonated credentials if that wasn't clear.